critiquebrainz/frontend/static/scripts/rating.js

```diff
@@ -0,0 +1,157 @@
+(function ($) {
+  'use strict';
+
+  var clearClass = 'rating-clear',
+    clearSelector = '.' + clearClass,
+    hiddenClass = 'hidden',
+    DEFAULTS = {
+      'min': 20,
+      'max': 100,
+      'step': 20,
+      'title': ["Terrible", "Bad", "Average", "Good", "Extraordinary"],
+      'empty-value': null,
+      'iconLib': 'glyphicon',
+      'activeIcon': 'glyphicon-star',
+      'inactiveIcon': 'glyphicon-star-empty',
+      'color': '#DAA520',
+      'clearable': true,
+      'clearableIcon': 'glyphicon-minus-sign',
+      'clearableRemain': false,
+      'inline': false,
+      'readonly': false
+    };
+
+  function starSelector(value) {
+    return '[data-value' + (value ? ('=' + value) : '') + ']';
+  }
+
+  function toggleActive($el, active, options) {
+    var activeClass = options.activeIcon,
+      inactiveClass = options.inactiveIcon;
+    $el.removeClass(active ? inactiveClass : activeClass).addClass(active ? activeClass : inactiveClass);
+  }
+
+  function parseOptions($input, options) {
+    var data = $.extend({}, DEFAULTS, $input.data(), options);
+    data.inline = data.inline === '' || data.inline;
+    data.readonly = data.readonly === '' || data.readonly;
+    data.clearable = data.clearable === '' || data.clearable;
+    return data;
+  }
+
+  function createRatingEl($input, options) {
+    // Inline option
+    if (options.inline) {
+      var $ratingEl = $('<span class="rating-input"></span>');
+    } else {
+      var $ratingEl = $('<div class="rating-input"></div>');
+    }
+
+    // Copy original classes but the rating class
+    $ratingEl.addClass($input.attr('class'));
+    $ratingEl.removeClass('rating');
+
+    // Render rating icons
+    for (var i = options.min; i <= options.max; i += options.step) {
+      $ratingEl.append('<i class="' + options.iconLib + '" data-value="' + i + '" title="' + options.title[i/20-1] + '" style="color:' + options.color + '"></i>');
+    }
+
+    // Render clear link
+    if (options.clearable && !options.readonly) {
+      $ratingEl.append('&nbsp;').append(
+        '<a class="' + clearClass + '">' +
+          '<i class="' + options.iconLib + ' ' + options.clearableIcon + '" />' +
+        '</a>'
+      );
+    }
+    return $ratingEl;
+  }
+
+  var Rating = function(input, options) {
+    var $input = this.$input = input;
+    this.options = parseOptions($input, options);
+    var $ratingEl = this.$el = createRatingEl($input, this.options);
+    $input.addClass(hiddenClass).before($ratingEl);
+    $input.attr('type', 'hidden');
+    this.highlight($input.val());
+  };
+
+  Rating.VERSION = '0.4.0';
+
+  Rating.DEFAULTS = DEFAULTS;
+
+  Rating.prototype = {
```

Comment:
Would it make sense to use [Classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) here?